tests/mysql/mysql.rs

```diff
@@ -39,6 +39,64 @@ async fn it_connects_without_password() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[sqlx_macros::test]
+async fn it_connects_with_zlib_compression() -> anyhow::Result<()> {
+    let url = Url::parse(&env::var("DATABASE_URL").context("expected DATABASE_URL")?)
+        .context("error parsing DATABASE_URL")?;
+    let mut conn = MySqlConnectOptions::from_url(&url)?
+        .compression(Compression::Zlib.default())
+        .connect()
+        .await?;
+
+    let rows = sqlx::raw_sql(r#"SHOW SESSION STATUS LIKE 'Compression'"#)
+        .fetch_all(&mut conn)
+        .await?;
+
+    let result = rows
+        .first()
+        .map(|r| r.try_get::<String, _>(1).unwrap_or_default())
+        .unwrap_or_default();
+
+    assert!(!rows.is_empty());
+    assert_eq!(result, "ON");
+
+    Ok(())
+}
+
+#[sqlx_macros::test]
+#[cfg(all(
+    not(any(
+        mariadb = "verylatest",
+        mariadb = "10_6",
+        mariadb = "10_11",
+        mariadb = "11_4",
+        mariadb = "11_8",
+    )),
+    feature = "mysql"
+))]
```

Comment:
This can actually just be `#[cfg(not(mariadb))]`.

The file itself requires `feature = "mysql"` and specifying individual versions is going to be too difficult to keep up to date.